lib/matplotlib/backends/_backend_tk.py

```diff
@@ -519,7 +519,10 @@ class NavigationToolbar2Tk(NavigationToolbar2, tk.Frame):
         If you want to use the toolbar with a different layout manager, use
         ``pack_toolbar=False``.
     """
-    def __init__(self, canvas, window, *, pack_toolbar=True):
+
+    def __init__(self, canvas, window=None, *, pack_toolbar=True):
+        if window is None:
+            window = canvas.get_tk_widget().master
         # Avoid using self.window (prefer self.canvas.get_tk_widget().master),
         # so that Tool implementations can reuse the methods.
         self.window = window
```

Comment:
Based on this, shouldn't `self.window` be deprecated as well?

Reply:
Sure.